test/threadstest.c

```diff
@@ -600,20 +600,19 @@ static int test_multi_load(void)
         res = test_multi_default();
     }
 
-    if (!TEST_true(thread_setup_libctx(1, NULL)))
-        return 0;
-
     /*
      * Check the legacy provider is loadable. If its not we assume it is
      * disabled and skip this test
      */
-    prov = OSSL_PROVIDER_load(multi_libctx, "legacy");
-    if (!TEST_ptr(prov))
-        return res;
+    prov = OSSL_PROVIDER_load(NULL, "legacy");
+    if (prov == NULL)
+        goto end;
```

Comment:
Does a TEST_error() call in the body cause the test to stop working?  This pushes the function call to after the problem has happened rather than as part of the detection of the problem.

Reply:
I read your comment several times, and I'm still not sure I understand what you are asking. Can you clarify?

Reply:
Perhaps better:

Add a TEST_error() call to output a message in this failure condition.
It shouldn't impact the timing of the code.

Reply:
This is not an error. It can legitimately happen in the event of no-legacy so it does not seem appropriate to use `TEST_error`. I could use `TEST_info` instead.

Reply:
> I could use TEST_info instead.

Done

Reply:
You could actually use default provider as fallback in the test in the no-legacy build.

Reply:
Good idea. Done.